src/guide/customization.rs

```diff
@@ -0,0 +1,398 @@
+//! A tutorial on how to customize eyre by overriding the Context parameter
+//!
+//! # Introduction
+//!
+//! In this tutorial we will go through the process of implementing a custom context step by step:
+//!
+//! * Start with defining an empty context that captures nothing
+//! * Add support for capturing backtraces
+//!     * skip backtrace capture if our source captured one
+//! * Add support for a http status codes
+//!     * Implement a setter for Result<T, Report> with a new Trait
+//!
+//! # A Basic Context
+//!
+//! The simplest context is an empty context, so lets start there, here's what our context looks
+//! like:
+//!
+//! ```rust
+//! struct CustomContext;
+//! ```
+//!
+//! Now this context doesn't add much context, but it does have the advantage of removing context,
+//! aka the backtrace, that may be unnecessary. To use this context we need to implement
+//! `EyreContext`. This provides the necessary interface for eyre to construct and use your context.
+//!
+//! First, we have to impl `default` so eyre knows how to construct your context when it creates
+//! your `Report` from a `std::error::Error`:
+//!
+//!
+//! ```rust
+//! fn default(_: &(dyn std::error::Error + 'static)) -> Self {
+//!     Self
+//! }
+//! ```
+//!
+//! And next we need to implement `debug` to tell eyre how to format the final report including
+//! your custom context:
```

Comment:
I think it'd be good to clarify why this method is called "debug", and how it differs from `fmt::Debug`, perhaps with an intra-doc link to the documentation on the `EyreContext` trait.